src/main/java/org/alfresco/rest/api/impl/GroupsImpl.java

```diff
@@ -809,7 +810,15 @@ public void deleteGroupMembership(String groupId, String groupMemberId)
         }
 
         validateGroupMemberId(groupMemberId);
-        // TODO: Verify if groupMemberId is member of groupId
+
+        // Verify if groupMemberId is member of groupId
+        AuthorityType authorityType = AuthorityType.getAuthorityType(groupMemberId);
+        Set<String> parents = authorityService.getContainingAuthorities(AuthorityType.GROUP, groupMemberId, true);
+        if (!parents.contains(groupId))

```

Comment:
Eventually used getContainingAuthorities to get only parent associations of the groupMemberId instead of using getContainedAuthority that returns all children. Digged deeper into the code and reviewed several performance issues in this area and it seems that the performance problems appeared in case of user authorities and the optimization using bridge table was made for that case.
Also regarding the usage of getContainingAuthorities noticed that this is used in other areas to do the check of membership (I overmissed that).

Reply:
It would be good to capture your reasoning for the change from `getContainedAuthorities()` to `getContainingAuthorities()` (rather than the bridge table). Is it that the assumption is `getContainingAuthorities()` will return far fewer results since we expect someone to belong to a small number of groups compared to the number of people that may be in a group (and would therefore be returned by `getContainedAuthorities()`)?

If so, it sounds like a fair assumption, so this should be ok.

Reply:
Yes, that is the reason and beside that I've seen this is the method used in other areas to check group membership.